tests/scripts/thread-cert/run_cert_suite.py

```diff
@@ -151,48 +151,48 @@ def release(self, port_offset: int):
         self._pool.put_nowait(port_offset)
 
 
+def print_summary(scripts: List[str], script_successes: Dict[str, List[int]], script_failures: Dict[str, List[int]]):
+    print("---------------------------------------")
+    print("Summary")
+    print("---------------------------------------")
+    for script in scripts:
+        success_count = len(script_successes[script])
+        failure_count = len(script_failures[script])
+        color = _COLOR_PASS if failure_count == 0 else _COLOR_FAIL
+        message = f'{color}PASS {success_count} FAIL {failure_count}{_COLOR_NONE} {script}'
+        if failure_count > 0:
+            message += f' {_COLOR_FAIL}Failed iterations: {script_failures[script]}{_COLOR_NONE}'
+        print(message)
+
+
 def run_tests(scripts: List[str], multiply: int = 1, run_directory: str = None):
-    script_fail_count = Counter()
-    script_succ_count = Counter()
+    scripts = list(set(scripts))
 
     # Run each script for multiple times
     script_ids = [(script, i) for script in scripts for i in range(multiply)]
     port_offset_pool = PortOffsetPool(MAX_JOBS)
 
-    def error_callback(port_offset, script, err, start_time):
-        port_offset_pool.release(port_offset)
+    # From the test script path to the iteration IDs
+    script_failures: Dict[str, List[int]] = defaultdict(list)
+    script_successes: Dict[str, List[int]] = defaultdict(list)
 
-        elapsed_time = round(time.time() - start_time)
-        script_fail_count[script] += 1
-        if script_succ_count[script] + script_fail_count[script] == multiply:
-            color = _COLOR_PASS if script_fail_count[script] == 0 else _COLOR_FAIL
-            print(
-                f'{color}PASS {script_succ_count[script]} FAIL {script_fail_count[script]}{_COLOR_NONE} {script} in {elapsed_time}s'
-            )
-
-    def pass_callback(port_offset, script, start_time):
+    def result_callback(script_id, script, dic, port_offset):
```

Comment:
Does `script_id` here mean the iteration id? Maybe use `iteration_id`?

Reply:
Changed all occurrences to `iteration_id`.